core/App/ehsm_core_test.cpp

```diff
@@ -807,168 +1635,482 @@ void test_generate_datakey()
-    printf("NAPI_CreateKey CMK with RSA SUCCESSFULLY!\n");
+    printf("GenerateDataKeyWithoutPlaintext_Json = %s\n", returnJsonChar);
+
+    ciphertext_without_base64 = retJsonObj.readData_cstr("ciphertext");
+    printf("GenerateDataKeyWithoutPlaintext SUCCESSFULLY!\n");
 
-    /* export the datakey with the new user public key */
-    returnJsonChar = NAPI_ExportDataKey(cmk_base64, ukey_base64, input_aad_base64.c_str(), olddatakey_base64);
+    payload_json.addData_string("ciphertext", ciphertext_without_base64);
+
+    param_json.addData_uint16("action", EH_DECRYPT);
+    param_json.addData_JsonValue("payload", payload_json.getJson());
+
+    returnJsonChar = EHSM_NAPI_CALL((param_json.toString()).c_str());
     retJsonObj.parse(returnJsonChar);
-    if(retJsonObj.getCode() != 200){
-        printf("NAPI_ExportDataKey failed, error message: %s \n", retJsonObj.getMessage().c_str());
+    if (retJsonObj.getCode() != 200)
+    {
+        printf("Failed to Decrypt the data, error message: %s \n", retJsonObj.getMessage().c_str());
         goto cleanup;
     }
-    printf("NAPI_ExportDataKey Json : %s\n", returnJsonChar);
-    printf("NAPI_ExportDataKey SUCCESSFULLY!\n");
+    printf("step2 Decrypt_Json = %s\n", returnJsonChar);
+    printf("Decrypt step2 data SUCCESSFULLY!\n");
+
+cleanup:
+    SAFE_FREE(ciphertext_without_base64);
+    SAFE_FREE(ciphertext_base64);
+    SAFE_FREE(cmk_base64);
+    SAFE_FREE(returnJsonChar);
+    printf("============test_generate_SM4_datakey end==========\n");
+}
+
+// /*
+
+// step1. generate an aes-gcm-128 key as the CM(customer master key)
+
+// step2. generate a cipher datakey without plaintext which encrypted by the CMK
+
+// step3. verify the cipher text could be decrypted by CMK correctly
+
+// step4. generate a new rsa key pair as the user-supplied asymmetric keymeterials.
+
+// step5. export the datakey with the new user public key
+
+// step6. verify that the new datakey cipher text could be decrypt succeed by the user rsa key pair
+
+void test_export_datakey()
+{
+    /*
+     * current testcase support aes-gcm-128, aes-gcm-192, aes-gcm-256, sm4-cbc, sm4-ctr cmk encrypted olddatakey
+     * export newdatakey using rsa2048, rsa3072, rsa4096, sm2 ukey
+     */
+    const char *keyspec_str[] =
+        {
+            "EH_AES_GCM_128",
+            "EH_AES_GCM_192",
+            "EH_AES_GCM_256",
+            "EH_RSA_2048",
+            "EH_RSA_3072",
+            "EH_RSA_4096",
+            "EH_EC_P224",
+            "EH_EC_P256",
+            "EH_EC_P384",
+            "EH_EC_P512",
+            "EH_HMAC",
+            "EH_SM2",
+            "EH_SM4_CTR",
+            "EH_SM4_CBC",
+            "INVALID_VALUE"};
+    ehsm_keyspec_t testcmkkeyspec[] = {EH_AES_GCM_128, EH_AES_GCM_192, EH_AES_GCM_256, EH_SM4_CBC, EH_SM4_CTR};
+    int testcmkkeyspecnum = 5;
+    ehsm_keyspec_t testukeykeyspec[] = {EH_RSA_2048, EH_RSA_3072, EH_RSA_4096, EH_SM2};
+    int testukeykeyspecnum = 4;
```

Comment:
not understand the name.  simplify and use _ to split them.

Reply:
> not understand the name. simplify and use _ to split them.

fixed, they are used to record cmk, ukey keyspec in this testcase.